plugin/externalPluginEntry.go

```diff
@@ -141,12 +187,101 @@ func (e *externalPluginEntry) ChildSchemas() []*EntrySchema {
 	return []*EntrySchema{}
 }
 
+const schemaFormat = `{
+	"type_id_one":{
+		"label": "one",
+		"methods": ["list"],
+		"children": ["type_id_two"]
+	},
+	"type_id_two":{
+		"label":"two",
+		"methods": ["read"]
+	}
+}
+`
+
 func (e *externalPluginEntry) Schema() *EntrySchema {
-	// TODO: Support external plugin schemas.
-	return nil
+	if !e.implements("schema") {
+		return nil
+	}
+	var graph *linkedhashmap.Map
+	if e.schemaGraphs != nil {
+		g, ok := e.schemaGraphs[e.typeID]
+		if !ok {
+			msg := fmt.Errorf(
+				"e.Schema(): entry schemas were prefetched, but no schema graph was provided for %v (%v)",
+				ID(e),
+				e.typeID,
+			)
+			panic(msg)
```

Comment:
Thinking this might be better if it were a warning.

Reply:
I'm ok with panic for now.

Reply:
Hmm. You start using log messages here, which we've avoided in plugins. They should use activity.

That suggests either:
- Schema needs to support returning an error (this makes sense to me specifically for external plugins)
- activity needs multiple log levels, and for warn/error we also expose that at the shell level.

I think both would be useful.

Reply:
Since we can operate without a schema, the least changes would be to add different levels to the activity log so we can have higher severity issues, and also log them appropriately.

Reply:
I can support the first option in the PR. Will update.

Reply:
(Documenting what we talked about on slack): Nevermind on the error handling support, reason for the warnings is b/c those are during development. Ideally once the external plugins finished, they'd prefetch their schema at the root so things would never error.